ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -693,6 +695,34 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
             logger.trace("{} Time used to find entry since: {}", partitionId,
                 (SystemTime.getInstance().milliseconds() - partitionStartTimeInMs));
 
+            // Compute CRC for backup recovery and verification
+            if (replicaMetadataRequestInfo.getReplicaPath().startsWith(BackupCheckerThread.DR_Verifier_Keyword)) {
+              Store storeToGet = storeManager.getStore(partitionId);
+              EnumSet<StoreGetOptions> storeGetOptions = EnumSet.of(StoreGetOptions.Store_Include_Deleted,
+                  StoreGetOptions.Store_Include_Expired);
+              List<MessageInfo> newMessageInfos = new ArrayList<>();
+              MessageFormatSend msgFmt = new MessageFormatSend();
+              // for-each blob
+              findInfo.getMessageEntries().forEach(minfo -> {
+                MessageInfo newMsgInfo = minfo;
+                try {
+                  List<StoreKey> keys = getConvertedStoreKeys(Collections.singletonList(minfo.getStoreKey()))
```

Comment:
we only have one storekey, we don't really need stream().distinct().collect. The returned list from getConvertedStoreKeys is totally fine.